v2/selective.go

```diff
@@ -103,19 +128,40 @@ func TraverseToFile(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, sele
 // TraverseV1 walks through the proposed dag traversal and writes a carv1 to the provided io.Writer
 func TraverseV1(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, selector ipld.Node, writer io.Writer, opts ...Option) (uint64, error) {
 	opts = append(opts, WithoutIndex())
+	conf := ApplyOptions(opts...)
 	tc := traversalCar{
-		size:     0,
+		size:     conf.DataPayloadSize,
 		ctx:      ctx,
 		root:     root,
 		selector: selector,
 		ls:       ls,
-		opts:     ApplyOptions(opts...),
+		opts:     conf,
 	}
 
-	len, _, err := tc.WriteV1(writer)
+	len, _, err := tc.WriteV1(tc.ctx, conf.SkipOffset, writer)
 	return len, err
 }
 
+// NewCarV1StreamReader creates an io.ReadSeeker that can be used to copy out the carv1 contents of a car.
+func NewCarV1StreamReader(ctx context.Context, ls *ipld.LinkSystem, root cid.Cid, selector ipld.Node, opts ...Option) (io.ReadSeeker, error) {
+	opts = append(opts, WithoutIndex())
+	conf := ApplyOptions(opts...)
+	tc := traversalCar{
+		size:     conf.DataPayloadSize,
+		ctx:      ctx,
+		root:     root,
+		selector: selector,
+		ls:       ls,
+		opts:     conf,
+	}
+	rwf := func(ctx context.Context, offset uint64, writer io.Writer) (uint64, error) {
+		s, _, err := tc.WriteV1(ctx, offset, writer)
```

Comment:
```suggestion
		// it's only at this point we have the `offset` to start writing at since the user of the
		// ReadSeeker will (may) have called Seek() and we've worked out where in the CAR
		// that is as an offset. Now we can start writing the CARv1 data, which will be passed
		// on to the ReadSeeker.
		// Note that we're inside a goroutine here
		s, _, err := tc.WriteV1(ctx, offset, writer)
```